timesketch/ui/static/js/controllers.js

```diff
@@ -104,21 +98,17 @@ timesketch.controller('TsExploreCtrl', function($scope, $http) {
         };
 
         $scope.saveView = function() {
-            var params = {
-                name: $scope.view_name,
-                query: $scope.query,
-                filter: $scope.filter
-            };
-            // TODO: Refactor out to a service.
-            $http.post('/api/v1/sketches/' + $scope.sketch_id + '/views/', params).success(function(data) {
-                var id = data.objects[0].id;
-                var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
-                window.location.href = view_url;
+            timesketchApi.saveView(
+                $scope.sketch_id, $scope.view_name, $scope.query, $scope.filter)
+                .success(function(data) {
+                    var id = data.objects[0].id;
+                    var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
+                    window.location.href = view_url;
             });
```

Comment:
It probably makes sense to handle requests failures in some generic way. For example, timesketchApi may expose a variable with the last error and then you can a piece of UI watching on it. Then, when error happens - no matter when or during which API call - you'll see it in the UI. This way you'll also won't have to manually handle failures everywhere.